backend_new/src/services/application.service.ts

```diff
@@ -195,68 +211,343 @@ export class ApplicationService {
+      ) {
+        // if the submission is after the application due date
+        throw new BadRequestException(
+          `Listing is not open for application submission`,
+        );
+      }
+    }
+
+    const rawApplication = this.prisma.applications.create({
+      data: {
+        ...dto,
+        confirmationCode: this.generateConfirmationCode(),
+        applicant: dto.applicant
+          ? {
+              create: {
+                ...dto.applicant,
+                applicantAddress: {
+                  create: {
+                    ...dto.applicant.applicantAddress,
+                  },
+                },
+                applicantWorkAddress: {
+                  create: {
+                    ...dto.applicant.applicantWorkAddress,
+                  },
+                },
+              },
+            }
+          : undefined,
+        accessibility: dto.accessibility
+          ? {
+              create: {
+                ...dto.accessibility,
+              },
+            }
+          : undefined,
+        alternateContact: dto.alternateContact
+          ? {
+              create: {
+                ...dto.alternateContact,
+                address: {
+                  create: {
+                    ...dto.alternateContact.address,
+                  },
+                },
+              },
+            }
+          : undefined,
+        applicationsAlternateAddress: dto.applicationsAlternateAddress
+          ? {
+              create: {
+                ...dto.applicationsAlternateAddress,
+              },
+            }
+          : undefined,
+        applicationsMailingAddress: dto.applicationsMailingAddress
+          ? {
+              create: {
+                ...dto.applicationsMailingAddress,
+              },
+            }
+          : undefined,
+        listings: dto.listings
+          ? {
+              connect: {
+                id: dto.listings.id,
+              },
+            }
+          : undefined,
+        demographics: dto.demographics
+          ? {
+              create: {
+                ...dto.demographics,
+              },
+            }
+          : undefined,
+        preferredUnitTypes: dto.preferredUnitTypes
+          ? {
+              connect: dto.preferredUnitTypes.map((unitType) => ({
+                id: unitType.id,
+              })),
+            }
+          : undefined,
+        householdMember: dto.householdMember
+          ? {
+              create: dto.householdMember.map((member) => ({
+                orderId: member.orderId,
+                firstName: member.firstName,
+                middleName: member.middleName,
+                lastName: member.lastName,
+                birthMonth: member.birthMonth,
+                birthDay: member.birthDay,
+                birthYear: member.birthYear,
+                sameAddress: member.sameAddress,
+                relationship: member.relationship,
+                workInRegion: member.workInRegion,
```

Comment:
is there any reason we can't do `...member` here instead of manually adding each one?

Reply:
I was having some issues with typescript but after looking at it again I realized what the issue was, should be simplified